### PR TITLE
kvserver,logstore: move log appends to logstore

### PR DESCRIPTION
**Pull request by pav-kv**
This change moves a few components related to storing the newly appended Raft entries from `kvserver` to the `logstore` package.

Part of #91979
Release note: None

### CODE REVIEW

**Review by tbg — APPROVED** [review]
image

File: pkg/kv/kvserver/logstore/logstore.go [hunk 1]
```diff
@@ -127,3 +142,94 @@ type AppendStats struct {
...
+		// changes, we must write and sync the Entries before the HardState.
+		if err := s.StateLoader.SetHardState(ctx, batch, rd.HardState); err != nil {
+			const expl = "during setHardState"
+			return RaftState{}, errors.Wrap(err, expl)
+		}
+	}
+	// Synchronously commit the batch with the Raft log entries and Raft hard
+	// state as we're promising not to lose this data.
+	//
+	// Note that the data is visible to other goroutines before it is synced to
```
[the remark this answers is not in the document]
Comment (pavelkalinnikov) [reply pkg/kv/kvserver/logstore/logstore.go:154]: The commit that introduced this comment: 82cbb49f.

[the remark this answers is not in the document]
Comment (pavelkalinnikov) [reply pkg/kv/kvserver/logstore/logstore.go:154]: Not sure if it was a drive-by comment at the time.

[the remark this answers is not in the document]
Comment (pavelkalinnikov) [reply pkg/kv/kvserver/logstore/logstore.go:154]: I added a TODO to revisit this comment later. It seems to be in a wrong place now, `logserver` is now less responsible for maintaining Raft invariants.

**Comment by pavelkalinnikov** [comment]
bors r=tbg

**Pull request merged**